cmd/tk/workflow.go

```diff
@@ -137,6 +137,16 @@ func showCmd() *cobra.Command {
 	}
 	vars := workflowFlags(cmd.Flags())
 	cmd.Run = func(cmd *cobra.Command, args []string) {
+		fi, err := os.Stdout.Stat()
+		if err != nil {
+			log.Fatalln("stdout pipe detection:", err)
+		}
+
+		if fi.Mode()&os.ModeCharDevice == 0 {
+			fmt.Fprintf(os.Stderr, "⚠️  Do not pipe the output of tk show. Use tk apply instead ⚠️\n")
+			return
```

Comment:
There should be a way to circumvent the safe-guard.

For example an ugly flag like `--dangerous-allow-piping` which is `false` by default

Reply:
Done. I went with `--dangerous-allow-redirect` because this check applies for both `tk show ... | cat` and `tk show ... > output.txt`. I tried seeing if there was a way to only have it work for piping but that doesn't seem to be possible.